nc/include/component/Registry.h

```diff
@@ -155,32 +145,25 @@ namespace nc::ecs
             void Clear();
 
         private:
-            storage_type m_storage;
+            std::vector<std::unique_ptr<PerComponentStorageBase>> m_registeredStorage;
             std::vector<Entity> m_active;
             std::vector<Entity> m_toAdd;
             std::vector<Entity> m_toRemove;
             HandleManager m_handleManager;
+            size_t m_maxEntities;
 
             template<Component T>
-            auto GetStorageFor() -> PerComponentStorage<T>& { return std::get<PerComponentStorage<T>>(m_storage); }
+            inline static PerComponentStorage<T>* m_typedStoragePtr = nullptr;
```

Comment:
And this works? This is pretty neat...

Reply:
Yes. Compiling the project with a new type instantiates the new template, but because it is static, it doesn't affect the binary layout of the Registry object. No need to recompile any engine source.